presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -84,6 +84,11 @@ public TableType getTableType()
         return snapshotId;
     }
 
+    public void setSnapshotId(Optional<Long> snapshotId)
```

Comment:
Handles should be immutable. Update this by creating a new object with the updated value.

Reply:
We don't really even need a new handle here. We simply need to send in the retrieved snapshot-id.